sycl/source/detail/platform_impl.cpp

```diff
@@ -15,12 +15,34 @@
 
 #include <algorithm>
 #include <cstring>
+#include <map>
 #include <regex>
 
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace detail {
 
+static std::map<RT::PiDevice, std::shared_ptr<device_impl>> device_impl_map;
+static std::mutex device_impl_map_mutex;
+static device host_device;
```

Comment:
https://llvm.org/docs/CodingStandards.html#do-not-use-static-constructors

Reply:
Please, avoid using global objects with non-trivial c'tor and d'tor.

Reply:
I fixed this as best as I could, @romanovvlad